jump-core/src/main/java/com/bitdecay/jump/BitWorld.java

```diff
@@ -179,6 +236,57 @@ private void resolveAndApplyPendingResolutions() {
 		pendingResolutions.clear();
 	}
 
+	private void expireContact(BitBody body) {
+		Iterator<BitBody> iterator = contacts.get(body).iterator();
+		BitBody otherBody = null;
+		while(iterator.hasNext()) {
+			otherBody = iterator.next();
+			if (GeomUtils.intersection(body.aabb, otherBody.aabb) == null) {
+				iterator.remove();
+				for (ContactListener listener : body.getContactListeners()) {
+					listener.contactEnded(otherBody);
+				}
+				for (ContactListener listener : otherBody.getContactListeners()) {
+					listener.contactEnded(body);
+				}
+			}
+		}
+	}
+
+	private void findNewContact(BitBody body) {
+		// TODO: collide vs Static and Dynamic dynamicBodies and report them via a callback (preferably)
+		// We need to update each body against the level grid so we only collide things worth colliding
+		BitPoint startCell = body.aabb.xy.floorDivideBy(tileSize, tileSize).minus(gridOffset);
```

Comment:
I've been thinking about it for a while, and at first I was opposed to coupling the physics system with the tile based level  stuff, but I think I've come around.  I think we just need to lean into it more.  If the physics system is coupled with a specific level layout, then it needs to be impossible to describe a physics world without that level information and it should complain at compile time, not runtime.